src/tests/suite/daos_test_common.c

```diff
@@ -898,7 +899,10 @@ daos_kill_server(test_arg_t *arg, const uuid_t pool_uuid,
 		      "disabled, svc->rl_nr %d)!\n", rank, arg->srv_ntgts,
 		       arg->srv_disabled_ntgts - 1, svc->rl_nr);
 
-	rc = daos_mgmt_svc_rip(grp, rank, true, NULL);
+	/* build and invoke dmg cmd to stop the server */
+	snprintf(dmg_cmd, sizeof(dmg_cmd),
```

Comment:
You need to check the return code.   Alternatively, you could use dts_create_config here instead and make buffer use DTS_CFG_MAX as the size.   This function checks the return code and ensures the string is null terminated.

Reply:
Admittedly, this function is poorly named for what you're doing...but it does the same thing